core-services/workflow/src/main/java/org/silverpeas/core/workflow/engine/TaskSavedRequest.java

```diff
@@ -0,0 +1,115 @@
+package org.silverpeas.core.workflow.engine;
+
+import org.silverpeas.core.persistence.Transaction;
+import org.silverpeas.core.util.Mutable;
+import org.silverpeas.core.util.ServiceProvider;
+import org.silverpeas.core.workflow.api.ProcessInstanceManager;
+import org.silverpeas.core.workflow.api.WorkflowException;
+import org.silverpeas.core.workflow.api.event.TaskSavedEvent;
+import org.silverpeas.core.workflow.api.instance.UpdatableHistoryStep;
+import org.silverpeas.core.workflow.api.instance.UpdatableProcessInstance;
+import org.silverpeas.core.workflow.api.model.State;
+import org.silverpeas.core.workflow.engine.instance.HistoryStepImpl;
+import org.silverpeas.core.workflow.engine.instance.ProcessInstanceImpl;
+import org.silverpeas.core.workflow.engine.model.StateImpl;
+
+/**
+ * A TaskSaved indicates the workflow engine that a task has been saved
+ */
+class TaskSavedRequest extends AbstractRequest {
+
+  private TaskSavedEvent event;
+
+  protected TaskSavedRequest() {
+  }
+
+  public static TaskSavedRequest get(final TaskSavedEvent event) {
+    TaskSavedRequest request = ServiceProvider.getService(TaskSavedRequest.class);
+    request.event = event;
+    return request;
+  }
+
+  @Override
+  public void process(final Object context) throws InterruptedException {
+
+    // Get the process instance
+    UpdatableProcessInstance instance = (UpdatableProcessInstance) event.getProcessInstance();
+    String id = instance.getInstanceId();
+    ProcessInstanceManager instanceManager = WorkflowHub.getProcessInstanceManager();
+    Mutable<UpdatableHistoryStep> step = Mutable.of(null);
+
+    Transaction.performInOne(()-> {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce this lambda expression number of lines from 21 to at most 20. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1188)